synapseclient/models/mixins/json_schema.py

```diff
@@ -0,0 +1,221 @@
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, AsyncGenerator, List, Optional
+
+from synapseclient.api.json_schema_services import (
+    bind_json_schema_to_entity,
+    delete_json_schema_from_entity,
+    get_invalid_json_schema_validation,
+    get_json_schema_derived_keys,
+    get_json_schema_from_entity,
+    get_json_schema_validation_statistics,
+    validate_entity_with_json_schema,
+)
+from synapseclient.core.async_utils import async_to_sync
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+
+@dataclass
+class JsonSchemaVersionInfo:
+    organizationId: str
+    organizationName: str
+    schemaId: str
+    schemaName: str
+    versionId: str
+    schema_id: str  # renamed from "$id" to a valid Python identifier
+    semanticVersion: str
+    jsonSHA256Hex: str
+    createdOn: str  # ISO datetime
+    createdBy: str  # synapse user ID
+
+
+@dataclass
+class JSONSchemaBindingResponse:
+    jsonSchemaVersionInfo: JsonSchemaVersionInfo
+    objectId: int
+    objectType: str
+    createdOn: str  # ISO datetime
+    createdBy: str  # synapse user ID
+    enableDerivedAnnotations: bool
+
+
+@dataclass
+class JSONSchemaValidationResponse:
+    objectId: str  # Synapse ID of the object (e.g., "syn12345678")
+    objectType: str  # Type of the object (e.g., "entity")
+    objectEtag: str  # ETag of the object at the time of validation
+    schema_id: str  # Renamed from "schema$id" to a valid Python identifier
+    isValid: bool  # True if the object content conforms to the schema
+    validatedOn: str  # ISO 8601 timestamp of when validation occurred
+
+
+@dataclass
+class JSONSchemaValidationStatisticsResponse:
+    containerId: str  # Synapse ID of the parent container
+    totalNumberOfChildren: int  # Total number of child entities
+    numberOfValidChildren: int  # Number of children that passed validation
+    numberOfInvalidChildren: int  # Number of children that failed validation
+    numberOfUnknownChildren: int  # Number of children with unknown validation status
+
+
+@dataclass
+class CausingException:
+    keyword: str
+    pointerToViolation: str
+    message: str
+    schemaLocation: str
+    causingExceptions: List["CausingException"] = field(default_factory=list)
+
+
+@dataclass
+class ValidationException:
+    pointerToViolation: str
+    message: str
+    schemaLocation: str
+    causingExceptions: List[CausingException]
```

Comment:
These fields will eventually need to be moved to `snake_case`. This is one of the small conversion items that makes working with the rest APIs a bit annoying. Since `snake_case` is the standard in Python, but `camelCase` is the standard in Java it makes for a small bit of work on our side.

This is the purpose of these `fill_from_dict` functions:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/fa4e17c4e7beef9cd5203b5fd5497ae4a38d6e18/synapseclient/models/materializedview.py#L669-L700

It's also similar to why when we go in the other direction (To store data in Synapse) we have these `to_synapse_request` functions:

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/fa4e17c4e7beef9cd5203b5fd5497ae4a38d6e18/synapseclient/models/materializedview.py#L702-L728